sys/arm64/arm64/trap.c

```diff
@@ -295,6 +295,7 @@ data_abort(struct thread *td, struct trapframe *frame, uint64_t esr,
 static void
 print_registers(struct trapframe *frame)
 {
+#if !__has_feature(capabilities)
```

Comment:
Given the kernel is missing large amounts of features I would not include this diff and just leave it broken to fix in a later commit.

Reply:
It won't compile if we do that. I preferred casts+TODO. We have to ifdef it out if we don't change the format strings.

Reply:
Or pragmas to fix the -Wformat warnings if that's the issue?

Reply:
As long as there is a diff to upstream that reminds us to implement this, I don't mind what we do for now.

Reply:
Well I would fix it to have the right code, and only use the pragma to deal with outdated toolchain issues.

Reply:
Yes that would obviously be better (fixes have been backported so no need for pragmas), but I don't think that needs to be part of this PR.

Reply:
See, I really don't like this break-a-function-just-so-it-compiles approach, it pollutes the history. It's fine for development, but the cleaned-up history that we merge in should not include such hacks, and just be a clean series of commits that get you from no CHERI to CHERI. Why can the single commit to fix the printing here not be a part of this PR? It should be independent from everything else.

Reply:
That's why I preferred the cast+TODO version since it would have printed useful output and compiled correctly.

Reply:
Yeah, not my preferred choice, but would be adequate in my eyes, since it's not _wrong_ in the sense that it's broken, insecure or less functional than baseline aarch64 (like some of the other things I've objected to before), just less helpful than it could be.

Reply:
... but also we don't have copyincap/copyoutcap yet so the kernel will fail to link, and thus talking about whether this file compiles or not is a bit pointless.

Reply:
I'm happy to switch back to casts + TODO.